includes/html/forms/update-ports.inc.php

```diff
@@ -19,9 +19,18 @@
 $rows_updated = 0;
 
 foreach ($_POST as $key => $val) {
+    $port_id = intval(substr($key, 7));
+    $port_group_ids = $_POST['port_group_' . $port_id];
+
+    dbDelete('port_group_port', '`port_id` = ?', [$port_id]);
```

Comment:
This should use the ->sync() method.  https://laravel.com/docs/8.x/eloquent-relationships#syncing-associations

Reply:
$_port = Port::where('port_id', '=', $port_id);
$_port->groups()->sync(port_group_ids);

won't work

also with port_group_ids:
[
[port_group_id => 2],
[port_group_id => 1],
[port_group_id => 3],
]


what do i wrong :S

Reply:
Use `array_values($port_group_ids)`  to get the correct format.

Reply:
Also, surrounding code here looks scary.

Reply:
surrounding code is not mine it's antique ;-)

```
use App\Models\Port;
$port_group_ids = $_POST['port_group_' . $port_id];
Port::find($port_id)->groups->sync(array_values($port_group_ids));

```

returns into:

```
[2021-02-09 23:40:55] production.ERROR: Call to a member function sync() on null {"userId":2,"exception":"[object] (Error(code: 0): Call to a member function sync() on null at /opt/librenms/includes/html/forms/update-ports.inc.php:34)
[stacktrace]
#0 /opt/librenms/html/ajax_form.php(26): include_once()
#1 {main}
"}

```
don't know why it is "null", port_id exists as primary key 

could use a hint :)

Reply:
it needs to be `groups()`.

->groups fetches the groups from the database and returns the collection.

->groups() access the relationship

Reply:
tried it with "groups()"  but it doesn't change anything

Reply:
@murrant 
tried it again, didn't get it to run

Reply:
`Device::find(1)->groups()->sync([1])` is a working example assuming device 1 and group 1 work.

Reply:
got it :D